### PR TITLE
Initial testing infrastructure for Spack

### PR DESCRIPTION
**Pull request by jjwilke**
This allows spack builds to be invoked instead of the usual tests.  Tests are invoked in exactly the same way except that the `--spack` option is given invokes Spack.

### CODE REVIEW

File: scripts/cm_test_all_sandia [hunk 1]
```diff
@@ -365,6 +412,8 @@ if [ "$MACHINE" = "sems" ]; then
                "cuda/9.2 $CUDA9_MODULE_LIST $CUDA_BUILD_LIST $KOKKOS_PATH/bin/nvcc_wrapper $CUDA_WARNING_FLAGS"
     )
   fi
+  SPACK_CUDA_ARCH="+maxwell50" #use an old one
```
Comment (ndellingwood) [inline scripts/cm_test_all_sandia]: Which sems machine has Maxwell GPUs to test? Just curious

Comment (jjwilke) [reply scripts/cm_test_all_sandia]: I don't know the specifics of any SEMS machine. I'm just guessing they have *at least* Maxwell 50.

Comment (ndellingwood) [reply scripts/cm_test_all_sandia]: Was just curious. Ready for merge then?

**Comment by jjwilke** [comment]
I think we can merge? I did some basic checks to make sure I didn't break anything that was already there. The existing tests, spot-check, and spot-check-tpls should all be maintained.

**Pull request merged**